server/expression/explicit_cast.go

```diff
@@ -111,6 +116,19 @@ func (c *ExplicitCast) Eval(ctx *sql.Context, row sql.Row) (any, error) {
 			return nil, err
 		}
 	}
+
+	if c.castToType.TypType == pgtypes.TypeType_Domain {
+		for _, check := range c.domainChecks {
+			res, err := sql.EvaluateCondition(ctx, check.Expr, sql.Row{})
```

Comment:
Would the row not contain the casted value?

Reply:
Right! Added a test for it.